File: PollfishIronSourceAdapterExampleObjC/PollfishIronSourceAdapterExampleObjC/ViewController.m
//
//  ViewController.m
//  PollfishIronSourceAdapterExampleObjC
//
//  Created by Fotis Mitropoulos on 31/3/22.
//

#import "ViewController.h"

// Place your app key here
#define APPKEY @"APP_KEY"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIButton *showRewardedAdBtn;
@end

@implementation ViewController

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    if (@available(iOS 14, *)) {
        [self requestIDFAPermission];
    } else {
        [self setupIronSource];
    }
}

- (void)requestIDFAPermission {
#if __has_include(<AppTrackingTransparency/AppTrackingTransparency.h>)
    if (@available(iOS 14, *)) {
        [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
            dispatch_async(dispatch_get_main_queue(), ^{
                [self setupIronSource];
            });
        }];
    } else {
        // Fallback on earlier versions
    }
#endif
}

- (void)setupIronSource {
    [ISIntegrationHelper validateIntegration];
    [IronSource initWithAppKey:APPKEY];
    [IronSource setRewardedVideoDelegate:self];
}

- (IBAction)showRewardedAd:(id)sender {
    [IronSource showRewardedVideoWithViewController:self];
}

- (void)rewardedVideoHasChangedAvailability:(BOOL)available {
    NSLog(@"%s", __PRETTY_FUNCTION__);
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.showRewardedAdBtn setEnabled:available];
    });
}

- (void)didReceiveRewardForPlacement:(ISPlacementInfo *)placementInfo {
    NSLog(@"%s %@ %@", __PRETTY_FUNCTION__, placementInfo.rewardAmount, placementInfo.rewardName);
}

- (void)rewardedVideoDidFailToShowWithError:(NSError *)error {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

- (void)rewardedVideoDidOpen {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

- (void)rewardedVideoDidClose {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

- (void)rewardedVideoDidStart {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

- (void)rewardedVideoDidEnd {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

- (void)didClickRewardedVideo:(ISPlacementInfo *)placementInfo {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

@end

File: PollfishIronSourceAdapterExampleSwift/PollfishIronSourceAdapterExampleSwift/ViewController.swift
//
//  ViewController.swift
//  PollfishIronSourceAdapterExampleSwift
//
//  Created by Fotis Mitropoulos on 31/3/22.
//

import UIKit
import ObjectiveC.runtime
import AppTrackingTransparency
#if canImport(AppTrackingTransparency)
import AppTrackingTransparency
#endif

class ViewController: UIViewController, ISRewardedVideoDelegate {
    
    private let appKey = "APP_KEY"
    @IBOutlet weak var showRewardedAdButton: UIButton!

    override func viewDidAppear(_ animated: Bool) {
        if #available(iOS 14, *) {
            requestIDFAPermission()
        } else {
            self.setupIronSource()
        }
    }
    
    @available(iOS 14, *)
    func requestIDFAPermission() {
        #if canImport(AppTrackingTransparency)
        ATTrackingManager.requestTrackingAuthorization { status in
            DispatchQueue.main.async {
                self.setupIronSource()
            }
        }
        #endif
    }

    private func setupIronSource() {
        ISIntegrationHelper.validateIntegration()
        
        IronSource.setRewardedVideoDelegate(self)
        IronSource.initWithAppKey(appKey)
    }
    
    @IBAction func showRewardedAd(_ sender: Any) {
        IronSource.showRewardedVideo(with: self)
    }
    
    public func didReceiveReward(forPlacement placementInfo: ISPlacementInfo!) {
        print(#function + " " + String(describing: placementInfo.self))
    }
    
    
    func rewardedVideoDidFailToShowWithError(_ error: Error!) {
        print(#function + " " + String(describing: error.self))
    }
    
    func rewardedVideoDidOpen() {
        print(#function)
    }
    
    func rewardedVideoDidClose() {
        print(#function)
    }
    
    func rewardedVideoDidEnd() {
        print(#function)
    }
    
    func rewardedVideoDidStart() {
        print(#function)
    }
    
    func rewardedVideoHasChangedAvailability(_ available: Bool) {
        showRewardedAdButton.isEnabled = available
        print(#function + " isAvailable: \(available)")
    }
    
    func didClickRewardedVideo(_ placementInfo: ISPlacementInfo!) {
        print(#function)
    }

}

